### PR TITLE
Make __bool__ raise for more cudf objects

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
To match pandas, this PR makes `DataFrame`, `MultiIndex` and `RangeIndex` raise on `__bool__`.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_dataframe.py [hunk 1]
```diff
@@ -11100,3 +11100,8 @@ def test_from_records_with_index_no_shallow_copy():
     data = np.array([(1.0, 2), (3.0, 4)], dtype=[("x", "<f8"), ("y", "<i8")])
     df = cudf.DataFrame(data.view(np.recarray), index=idx)
     assert df.index is idx
+
+
+def test_bool_raises():
+    with pytest.raises(TypeError):
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_dataframe.py:11106]: Can we use `assert_exceptions_equal` to validate and compare pandas errors? Same for below.

**Review by galipremsagar — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/tests/test_dataframe.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/tests/test_dataframe.py]: Sure thing. Changed

**Review by galipremsagar — APPROVED** [review]

File: python/cudf/cudf/tests/test_dataframe.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_dataframe.py]: Thanks!

**Pull request merged**